### PR TITLE
Fixing socket permissions on 4.15 kernels 

### PR DESCRIPTION
**Pull request by kubiko**
On 4.15 kernel, socket dir permissions needs to be rwk instead of rw otherwise directory creation is denied

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>

Thanks for helping us make a better snapd!
Have you signed the [license agreement](https://www.ubuntu.com/legal/contributors) and read the [contribution guide](https://github.com/snapcore/snapd/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

File: interfaces/builtin/avahi_observe.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ network netlink,
 
 # Allow access to daemon to create socket
 /{,var/}run/avahi-daemon/  w,
-/{,var/}run/avahi-daemon/{pid,socket} rw,
+/{,var/}run/avahi-daemon/{pid,socket} rwk,
```
Comment (bboozzoo) [inline interfaces/builtin/avahi_observe.go]: Something changed in avahi-daemon?

**Comment by kubiko** [comment]
I had impressions this was always wrong and it just happen to work on 4.4 kernels, but let's wait for @jdstrand comment

**Comment by kubiko** [comment]
@zyga @jdstrand any decision on this? Either we should make fix in 4.15 kernel or merge this change.
At the moment we don't seem to be doing either, so avahi interface is broken on core18 devices

**Comment by jdstrand** [comment]
FYI, since 2.38 branched I created https://github.com/snapcore/snapd/pull/6566

**Pull request merged**